Horizon/Horizon/Sources/Features/AI/AIFlashCard/View/AIFlashCardView.swift

```diff
@@ -0,0 +1,100 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2024-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+import Core
+
+struct AIFlashCardView: View {
+    @Bindable var viewModel: AIFlashCardViewModel
+    @Environment(\.viewController) private var viewController
+
+    var body: some View {
+        GeometryReader { geometry in
+            VStack {
+                headerView
+
+                ScrollView(.horizontal, showsIndicators: false) {
+                    LazyHStack {
+                        flashCardsView(geometry: geometry)
+                    }
+                    .scrollTargetLayout()
+                }
+                .scrollTargetBehavior(.viewAligned)
+                .contentMargins(.horizontal, 30, for: .scrollContent)
+                .scrollPosition(id: $viewModel.currentCardIndex)
+            }
+        }
+        .animation(.smooth, value: viewModel.currentCardIndex)
+        .paddingStyle(.top, .standard)
+        .safeAreaInset(edge: .bottom) {
+            FlashCardStepIndicatorView(viewModel: viewModel)
+                .padding(.bottom, 5)
+        }
+        .applyHorizonGradient()
+    }
```

Comment:
Let's rework this so we don't need to use `GeometryReader`. With `containerRelativeFrame` you can get the height in a closure, then do whatever is needed to calculate the height.

Reply:
Removed